src/code/z_parameter.c

```diff
@@ -1772,10 +1781,736 @@ void Interface_UpdateHudAlphas(PlayState* play, s16 dimmingAlpha) {
+            // (non-minigame?)
+            if ((player->stateFlags1 & PLAYER_STATE1_800000) && (player->currentMask == PLAYER_MASK_BLAST) &&
+                (gSaveContext.bButtonStatus == BTN_DISABLED)) {
+                // Riding Epona with blast mask?
+                restoreHudVisibility = true;
+                gSaveContext.bButtonStatus = BTN_ENABLED;
+            }
+
+            if (BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) != ITEM_NONE) {
+                if ((player->transformation == PLAYER_FORM_DEKU) && CHECK_WEEKEVENTREG(WEEKEVENTREG_08_01)) {
+                    gSaveContext.hudVisibilityForceButtonAlphasByStatus = true;
+                    if (play->sceneId == SCENE_BOWLING) {
+                        if (gSaveContext.buttonStatus[EQUIP_SLOT_B] == BTN_DISABLED) {
+                            gSaveContext.buttonStatus[EQUIP_SLOT_B] = BTN_ENABLED;
+                            gSaveContext.buttonStatus[EQUIP_SLOT_C_LEFT] = BTN_DISABLED;
+                            gSaveContext.buttonStatus[EQUIP_SLOT_C_DOWN] = BTN_DISABLED;
+                            gSaveContext.buttonStatus[EQUIP_SLOT_C_RIGHT] = BTN_DISABLED;
+                        }
+                    } else if (gSaveContext.buttonStatus[EQUIP_SLOT_B] == BTN_DISABLED) {
+                        gSaveContext.buttonStatus[EQUIP_SLOT_B] = BTN_ENABLED;
+                        gSaveContext.buttonStatus[EQUIP_SLOT_C_LEFT] = BTN_ENABLED;
+                        gSaveContext.buttonStatus[EQUIP_SLOT_C_DOWN] = BTN_ENABLED;
+                        gSaveContext.buttonStatus[EQUIP_SLOT_C_RIGHT] = BTN_ENABLED;
+                    }
+
+                    Interface_SetHudVisibility(HUD_VISIBILITY_B_MAGIC);
+                } else {
+                    if ((BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) != ITEM_BOW) &&
+                        (BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) != ITEM_BOMB) &&
+                        (BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) != ITEM_BOMBCHU)) {
+                        gSaveContext.hudVisibilityForceButtonAlphasByStatus = true;
+                        gSaveContext.buttonStatus[EQUIP_SLOT_B] = BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B);
+                        gSaveContext.buttonStatus[EQUIP_SLOT_C_LEFT] = BTN_ENABLED;
+                        gSaveContext.buttonStatus[EQUIP_SLOT_C_DOWN] = BTN_ENABLED;
+                        gSaveContext.buttonStatus[EQUIP_SLOT_C_RIGHT] = BTN_ENABLED;
+                        if (play->sceneId == SCENE_BOWLING) {
+                            if (CURRENT_DAY == 1) {
+                                BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) = ITEM_BOMBCHU;
+                            } else if (CURRENT_DAY == 2) {
+                                BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) = ITEM_BOMB;
+                            } else {
+                                BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) = ITEM_BOW;
+                            }
```

Comment:
I would guess they wanted it in one place so that you didn't get conflicting actors trying to set the button item. By having it grouped together, when this case reached, all subsequent cases are dropped. Only if every special case is not obtained is the general button-set in part2 applied. That's my interpretation. And I'm sure there could have been a better way as well